### PR TITLE
Fixing The Error with the WellData Class When Processing Operator Name Data

### PR DESCRIPTION
**Pull request by Jaffe11**
## Fixes/Addresses/Summary/Motivation:

This PR is to address the error that occurs when a user sets "ignore_operator_name" to False, but the class still requires a column for operator name. 

## Changes proposed in this PR:
- Moved the for loop for collecting rows with "unknown" as a value for operator name and placed it under the if statement that checks the value on the "ignore_operator_name" config option.
- Disabled Pylint warnings locally for TODO and FIXME statements in the well_data.py file.
- Added fixme to the list of words for Pylint to not spell check.
      -  This is Pylint's own flag for locally disabling TODO & FIXME warnings. 

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my
contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.md file
   at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has
   rights to intellectual property that includes these contributions, I represent that I have
   received permission to make contributions and grant the required license on behalf of that
   employer.

### CODE REVIEW

File: .pylintrc [hunk 1]
```diff
@@ -545,7 +545,7 @@ spelling-dict=en_US
 spelling-ignore-comment-directives=fmt: on,fmt: off,noqa:,noqa,nosec,isort:skip,mypy:
 
 # List of comma separated words that should not be checked.
-spelling-ignore-words=
+spelling-ignore-words=fixme
```
Comment (apd-ypuranik) [inline .pylintrc:548]: We already have a mechanism to mark words as 'correct' for spell check by adding them to this file. https://github.com/NEMRI-org/primo-optimizer/blob/main/.github/wordlist.txt

Please undo the change to pylintrc and instead add fixme to that file in alphabetical order

File: primo/data_parser/well_data.py [hunk 2]
```diff
@@ -37,6 +37,9 @@
 OWNER_WELL_COLUMN_NAME = "Owner Well-Count"
 
 
+# disabling too many lines in module warning
+# pylint: disable = C0302
```
Comment (apd-ypuranik) [inline primo/data_parser/well_data.py:41]: does too-many-lines as string not work? That is easier to interpret than having to check what C0302 means (although your comment above helps).

Comment (apd-ypuranik) [reply primo/data_parser/well_data.py:41]: You can also disable multiple pylint issues in one line as follows:
`#pylint: disable=too-many-lines, too-many-public-methods`

**Review by apd-ypuranik — CHANGES_REQUESTED** [review]

File: primo/data_parser/well_data.py [hunk 3]
```diff
@@ -991,8 +992,8 @@ def compute_priority_scores(self):
         # Check if all the required columns for supported metrics are specified
         # If yes, register the name of the column containing the data in the
         # data_col_name attribute
-        # TODO: Combine the check_columns_available method with this method.
-        # TODO: Check fill data consistency for ann_gas_production and
+        # TODO: Combine the check_columns_available method with this method.# pylint: disable=fixme
```
Comment (radhakrishnatg) [inline primo/data_parser/well_data.py:995]: Please remove these TODOs.

**Review by radhakrishnatg — APPROVED** [review]
Please make the changes Yash suggested. Otherwise, everything looks good to me.

File: primo/data_parser/well_data.py [hunk 4]
```diff
@@ -660,7 +663,7 @@ def _categorize_oil_gas(self):
             return
 
         # Ensure the required columns are present in the DataFrame
-        # FIXME: This value is being accepted as input at two different locations.
+        # FIXME: This value is being accepted as input at two different locations. # pylint: disable=fixme
```
[the remark this answers is not in the document]
Comment (apd-ypuranik) [reply primo/data_parser/well_data.py:666]: I think that is a great idea. I will take care of this in my PR :-)

File: .pylintrc (hunk 1 above)
Comment (Jaffe11) [reply .pylintrc]: Fixed

File: primo/data_parser/well_data.py [hunk 5]
```diff
@@ -991,8 +990,6 @@ def compute_priority_scores(self):
         # Check if all the required columns for supported metrics are specified
         # If yes, register the name of the column containing the data in the
         # data_col_name attribute
-        # TODO: Combine the check_columns_available method with this method.
-        # TODO: Check fill data consistency for ann_gas_production and
         # ann_oil_production. See _categorize_gas_oil_wells method for details.
```
Comment (strahlw) [inline primo/data_parser/well_data.py:994]: You should take away the complete TODO statement (also line 993). Are we sure that we want to get rid of these TODO statements because of pylint? I don't want to get rid of anything important.

**Review by strahlw — CHANGES_REQUESTED** [review]
Very minor request

**Review by apd-ypuranik — APPROVED** [review]

**Comment by apd-ypuranik** [comment]
@strahlw : Merge pending your approval :-)
